space-plugins/nuxt-base/server/utils/webhook.ts

```diff
@@ -1,20 +1,76 @@
 import { randomBytes } from 'node:crypto';
 import {
-	CreateStoryblokWebhook,
-	IsValidWebhookCreationParams,
+	SaveStoryblokWebhook,
+	DeleteStoryblokWebhook,
 	StoryblokWebhookResponse,
 	StoryblokWebhookEventCategory,
-	DeleteStoryblokWebhook,
+	IsValidWebhookSaveParams,
+	CreateStoryblokWebhook,
+	UpdateStoryblokWebhook,
 } from '../../types';
 
 export const generateSecret = (lengthInBytes: number = 20) => {
 	return randomBytes(lengthInBytes).toString('base64url');
 };
 
-export const createStoryblokWebhook: CreateStoryblokWebhook = async (
+export const createStoryblokWebhook: CreateStoryblokWebhook = async (params) =>
+	saveStoryblokWebhook('create', params);
+
+export const updateStoryblokWebhook: UpdateStoryblokWebhook = async (params) =>
+	saveStoryblokWebhook('update', params);
+
+export const deleteStoryblokWebhookById: DeleteStoryblokWebhook = async (
+	params,
+) => {
+	const apiHost = getManagementApiHost(params.spaceId);
+
+	const url = `${apiHost}/v1/spaces/${params.spaceId}/webhook_endpoints/${params.webhookId}`;
+
+	try {
+		await $fetch(url, {
+			headers: {
+				Authorization: `Bearer ${params.accessToken}`,
+			},
+			method: 'DELETE',
+		});
+
+		return {
+			ok: true,
+			result: 'Webhook deleted',
+		};
+	} catch (err: any) {
+		return {
+			ok: false,
+			error: 'could-not-delete-webhook',
+		};
+	}
+};
+
+export const inferEventCategoryFromBody = (
+	body: any,
+): StoryblokWebhookEventCategory | undefined => {
+	if (body.workflow_name) {
+		return 'workflow';
+	} else if (body.datasource_slug) {
+		return 'datasource';
+	} else if (body.release_id) {
+		return 'release';
+	} else if (body.branch_id) {
+		return 'pipeline';
+	} else if (body.user_id) {
+		return 'user';
+	} else if (body.asset_id) {
+		return 'asset';
+	} else if (body.story_id) {
+		return 'story';
+	}
+};
+
+const saveStoryblokWebhook: SaveStoryblokWebhook = async (
```

Comment:
nit: could we rename this to `upsertStoryblokWebhook`, it might be more fitting IMO.

Reply:
Thanks for the suggestion, Bibi.
I was trying to think of a better name but with no luck. 🤣

Reply:
Done Bibi https://github.com/storyblok/space-tool-plugins/pull/56/commits/93f0fb3d5434570f6db1e06b0ad110c43ffdcd60
🙌